DMCompiler/DM/Expressions/Builtins.cs

```diff
@@ -451,15 +451,27 @@ class Initial : DMExpression {
 
         public Initial(Location location, DMExpression expr) : base(location) {
             _expr = expr;
+            if (_expr is not LValue) {
+                // We do this check here on-construction,
+                // because our expression may poof into a non-LValue later on, after const folding.
+                DMCompiler.Error(Location, $"Cannot get initial value of {_expr}");
+            }
         }
 
         public override void EmitPushValue(DMObject dmObject, DMProc proc) {
             if (_expr is LValue lValue) {
                 lValue.EmitPushInitial(dmObject, proc);
                 return;
             }
+            _expr.EmitPushValue(dmObject, proc);
+        }
 
-            throw new CompileErrorException(Location, $"can't get initial value of {_expr}");
+        public override bool TryAsConstant(out Constant constant) {
```

Comment:
`initial()` doesn't appear to constant-fold in byond? This gives me `=: constant initializer required`

```
/var/const/a = 9
/var/const/b = initial(a)
```

Even if it did, I don't think it should use `_expr.TryAsConstant()`, as this should be able to constant-fold even on non-constant values because they always begin with the same value.

Reply:
I think it's a reasonable change to make `initial()` be const-folded. I think it better adheres to the user's expectation, since it's a proc that is *supposed* to be for querying compiletime values of things. I was quite surprised myself to hear that it wasn't const-folded). I also don't think it's a change that would ever negatively affect behaviour for any target, just make them run that little bit faster.

Rest of the review is on-point though, I'll look into that.